pquery-run.sh

```diff
@@ -1657,6 +1695,16 @@ pquery_test(){
         echoit "Saving full trial outcome (as SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY=0 and so trials are saved irrespective of whether an issue was detected or not)"
         savetrial
         TRIAL_SAVED=1
+      elif [[ ${SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY} -eq 1 && ${PQUERY3} -eq 1 ]]; then
+        if [ ${TRIAL} -gt 1 ]; then
+          echoit "Saving last trial outcome (as SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY=1)"
```

Comment:
remove SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY. only sufficient condition is PQUERY3 and remove 
echoit "Saving last trial outcome (as SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY=1)"

Reply:
We cannot exclude SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY
If you see the old pquery code, they were checking for this condition
 SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY -eq 0
If yes, then TRIAL_SAVED is set to 1, and by doing this removetrial function is not invoked
There was no check for SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY -eq 1, and the reason was, they had no intention to save anything
But now, we have an additional contraint that when SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY -eq 1, we have to not delete everything, but retain the last trial
Hence we need to have  a check in-place